pyccel/codegen/printing/ccode.py

```diff
@@ -209,7 +210,45 @@ def _print_PyccelNot(self, expr):
         return '!{}'.format(a)
 
     def _print_Import(self, expr):
-        return '#include "{0}"'.format(expr.source)
+        return '#include <{0}>'.format(expr.source)
+
+    def _print_String(self, expr):
+        return "\"{}\"".format(expr.arg)
+
+    def _print_Print(self, expr):
+        self._additional_imports.add("stdio.h")
+        args = []
+        args_format = []
+        end = '\\n'
+        sep = ' '
+        for f in expr.expr:
+            if isinstance(f, ValuedVariable):
+                if f.name == 'sep'      :   sep = str(f.value)
+                elif f.name == 'end'    :   end = str(f.value)
+            elif f.dtype is NativeBool():
+                args_format.append("%s")
+                args.append('({}) ? "True" : "False"'.format(self._print(f)))
+            elif f.dtype is NativeInteger():
+                args_format.append("%ld")
```

Comment:
`%ld` is for printing objects of type `long` if we use an `int32` or similar then we will have a different precision and we will no longer have a long

Reply:
I am confused now. As far as I know, we use long for `NativeInteger`.
I was thinking of using a `dtype_registry` with `arg_format`. but we use `arg_format` only in `_print_print`.
what do you suggest to use?

Reply:
yes, I think we always have a `long` but the definition of `long` changes from an architecture to another it's sometimes `int64` and sometimes `int32`

Reply:
yes, these changes will affect precision. but I think `printf` has nothing to do with that.

Reply:
I don't think that's right. I think we use long for (`NativeInteger`, precision=8). The default precision for integers is 8 (on most machines) so we almost always use `long`, however if we use numpy types then this is no longer the case. E.g. the following code:
```
@types('int16', 'int16')
def test_add(x, y):
    return x + y
```
is translated to:
```
short int test_add(short int x, short int y)
{
    short int Out_0001;
    Out_0001 = x + y;
    return Out_0001;
}
```
If you use `%ld` to print `x` then I think you should get a warning.

Now that PR #434 has been merged, if you merge the master branch in then you should be able to see this warning if you create tests using numpy types

Reply:
A dictionary sounds like a good idea to me. If it is only relevant for the function `_print_Print` then you can define it in that function rather than at the global scope

Reply:
In this case, we have another type that we should support with `%hd`.
I will try to improve the code format for easy including other types.

Reply:
You can take a look at the `dtype_registry` dictionary to see the currently implemented types. The first argument is one of `_print(NativeInteger())`, `_print(NativeReal())`, `_print(NativeBool())`, or `_print(NativeComplex())`, the second is the precision.

If you call pyccel with the argument `--flags="-Wall"` in your test then you can make sure that an error is raised if the format string is wrong